src/validators.js

```diff
@@ -71,3 +72,29 @@ export function validateVariableQuality(disableVariableQuality) {
     );
   }
 }
+
+export function validateTargetDPRRatios(targetDPRRatios) {
+  if (!Array.isArray(targetDPRRatios) || !targetDPRRatios.length) {
+    throw new Error(
+      'The targetDPRRatios argument can only be passed a valid non-empty array of integers',
+    );
+  } else {
+    const allValidDPR = targetDPRRatios.every(function (dpr) {
+      return DPR_TARGETS_RATIOS.includes(dpr);
+    });
+
+    if (!allValidDPR) {
+      throw new Error(
+        'The targetDPRRatios argument can only contains 1, 2, 3, 4 or 5 values',
+      );
+    }
+  }
```

Comment:
> I think this check is not necessary and should be removed because it stops the user defining custom dpr ratios (e.g. `1.5x`)

➕ 1, let's add a test for this in https://github.com/imgix/js-core/blob/961025bdce85c8ff1955a68ce5edddd815f5bdde/test/test-buildSrcSet.js#L724

Reply:
> I think this check is not necessary and should be removed because it stops the user defining custom dpr ratios (e.g. `1.5x`)

I'm working on review and I have an issue if user defining `1.5x` as custom DPR. We don't have quality on this dpr because it don't exist on `DPR_QUALITIES`

```js
        const CUSTOM_TARGETS_RATIOS = [1, 1.5, 2];

        const srcset = new ImgixClient({
          domain: 'testing.imgix.net',
          includeLibraryParam: false,
        }).buildSrcSet(
          'image.jpg',
          { w: 100 },
          { targetDPRRatios: CUSTOM_TARGETS_RATIOS },
        );

          console.log(srcset);
```

Print

```html
https://testing.imgix.net/image.jpg?w=100&dpr=1&q=75 1x,
https://testing.imgix.net/image.jpg?w=100&dpr=1.5 1.5x,
https://testing.imgix.net/image.jpg?w=100&dpr=2&q=50 2x
```

It is a problem for you ?

Reply:
I think if we did `Math.floor` on the dpr when searching through `DPR_QUALITIES`, that would work well. 

```js
    const withQuality = (path, params, dpr) => {
      return `${this.buildURL(path, {
        ...params,
        dpr: dpr,
        q: params.q || qualities[Math.floor(dpr)],
      })} ${dpr}x`;
    };
```

What do you think about this?

Reply:
Yes it's ok. 

Maybe `q: params.q || qualities[dpr] || qualities[Math.floor(dpr)]` to keep the right quality if user pass `{ 1.5: 40 }` to new `targetDPRRatiosQualities` params.

Reply:
Nice catch, that sounds good.